Documentation/adding-new-repository.md

```diff
@@ -0,0 +1,111 @@
+
+Adding a new repository requires changes in several places. This document aims to be a full recipe of all components and steps that require modifications.
+
+It is recommended to take a repository in the same domain (backend, client, etc.) as a model when going through the different phases.
+
+## GitHub: create repo and give access
+
+Create the new repository. Consult with the team, epic owner or product owner if the repository is to be open or private
+
+By default, we give write access for `developers` and `rndity` groups. Remove any individual user.
+
+ref https://github.com/mendersoftware/REPOSITORY/settings/access
+
+### Enterprise fork
+
+TODO
+
+## GitLab: project
+
+To craete the project in GitLab:
+
+1. Open an incognito window in your browser
+2. Log into GitHub as `mender-test-bot`, credentials are in `mystiko`
+3. Log into GitLab, using OAth from GitHub
+  * If asked for the token, use the one in `mystiko`
+4. New project >  Run CI/CD for external repository
+
+## GitLab: CI/CD pipeline
+
+Please use templates as much as possible. The standard is a bit different depending on the repository: backend repositories can be built using only templates, client repositories rely on `Makefile` and custom jobs to build and publish.
+
+## integration
+
+*NOTE*: needs to be done togeher with `mender-qa`.
+
+*Important*: changes in release_tool and yaml files need to go to integration/master, even when the repository is meant to be used only in a release branch while developing it.
+
+1. Add repo, image, and docker container to `component-maps.yml`
+  * ref https://github.com/mendersoftware/integration/blob/master/component-maps.yml
+  * While the repository is in development, set `release_component: false`
+  * For non-backend repositories, set `independent_component: true`
+2. Add repo to `git-versions.yml` and `git-versions-enterprise.yml` (where applicable)
+  * ref https://github.com/mendersoftware/integration/blob/master/git-versions.yml
+  * https://github.com/mendersoftware/integration/blob/master/git-versions-enterprise.yml
+3. Modify `release_tool.py`
+  * ref https://github.com/mendersoftware/integration/blob/master/extra/release_tool.py
+  1. Add repo to `GIT_TO_BUILDPARAM_MAP`
+  2. For backend repositories, add repo to `BACKEND_SERVICES_OPEN`, `BACKEND_SERVICES_ENT` or `BACKEND_SERVICES_OPEN_ENT`.
+  3. If new docker-compose file was added, add the file accordingly in `filter_docker_compose_files_list`.
+
+## mender-qa
+
+1. Modify Mender QA Pipeline:
+  * ref https://github.com/mendersoftware/mender-qa/blob/master/.gitlab-ci.yml
+  * Add new repo into `variables`
+  * Add new call to `checkout_repo` in `init_workspace`.
+    * For private repositories, use ssh Git url
+2. Modify `jenkins-yoctobuild-build.sh`
+  * ref https://github.com/mendersoftware/mender-qa/blob/master/scripts/jenkins-yoctobuild-build.sh
+  * Add the new Docker images, if applicable, to `Build server repositories.`
+
+## integration-test-runner
```

Comment:
One section you might want to add after this is:

```
## Docker images

Add Docker project on Docker hub, with correct permissions, etc, etc
```

Reply:
Very good catch, thanks.

Reply:
(I will add it before `GitLab: CI/CD pipeline`, as the image repo needs to be in place before the pipeline)

Reply:
I added:

```
## Docker image(s)

When applicable, prepare Docker image repositories for the new component.

* For Open Source repository, create a new public repository in Docker Hub. Teams and permissions:
  * admin: Admin
  * developers: Read-only
  * openrnd: Read-only
  * robots: Read & Write
* For closed source repository, create a new private repository in registry.mender.io
```